sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -288,6 +288,9 @@ private[hive] class SparkExecuteStatementOperation(
     if (statementId != null) {
       sqlContext.sparkContext.cancelJobGroup(statementId)
     }
+    if (result != null) {
+      result.destroy()
```

Comment:
Seems this is the entrance of resource cleaning. Could this cover the scenario of container killed?

Reply:
And if you change to `result.queryExecution.executedPlan.cleanUpResources()`, no need to add new API in `Dataset`

Reply:
I am not sure what you mean. You can check the discussion above for further details.

Reply:
Please ignore my first comment. I got it now. @mgaido91 . Why this PR closed?

Reply:
because an approach like the one proposed here https://github.com/apache/spark/pull/21502#issuecomment-397799870 is better, but I had no time to try it (and I also have a limited knowledge of that area of Spark).